tests/test_state.py

```diff
@@ -760,3 +760,32 @@ def _get_context(
 
         result = yield defer.ensureDeferred(self.state.compute_event_context(event))
         return result
+
+    def test_make_state_cache_entry(self):
+        "Test that calculating a prev_group and delta is correct"
+
+        new_state = {
+            ("a", ""): "E",
+            ("b", ""): "E",
+            ("c", ""): "E",
+        }
```

Comment:
how about a key that matches the old states, so we can check it is excluded from the delta?